Sally7/Internal/DisposableHelper.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace Sally7.Internal;
+
+internal static class DisposableHelper
+{
+    public static void ThrowIf(bool condition, object instance)
```

Comment:
Starting with .NET 7 theres [ObjectDisposedException.ThrowIf](https://learn.microsoft.com/en-us/dotnet/api/system.objectdisposedexception.throwif?view=net-8.0).

Should we `#if`def it here so the built-in method can be used?

Reply:
Not sure how we should go about declaring the static `ThrowIf` on net < 7? I took `ObjectDisposedException.ThrowIf` as example of course, just not sure how to create the compatibility here.

Reply:
Via a
```c#
private void EnsureNotDisposed()
{
#if NET7_OR_GREATER
    ObjectDisposedException.ThrowIf(_stream is null, this);
#else
    // Custom method
#endif
}
```
Or make that method `public` in the helper type.